### DISCUSSION TITLE
Is it possible to have a low memory version option? The memory usage is too high.

### DISCUSSION THREAD

**Discussion by binronn** [Flow-Launcher/Flow.Launcher#3138] [Q&A]
600+MB ！

**Comment by binronn** [comment]
I hardly use any third-party plugins, yet the memory usage is still very high.

**Reply by jjw24** [reply]
Hi there, we would need more details to determine the cause of such high memory:
1. Do you get this much when using only the plugins that come installed with flow? If not, what plugins have you installed and do you get the same high usage without them?
2. Does it immediately jump this high or after a particular action e.g. changed language or opened settings window or visited plugin store?
3. What is the windows version you are using?
4. Does the high memory drop after flow is not used for a while?

**Comment by binronn** [comment]
![image](https://github.com/user-attachments/assets/1ff93e36-5a00-4241-8824-a1101d9b95a4)
I used everything.

I haven't specifically paid attention to whether memory usage decreases over long periods of inactivity. From my impression, memory doesn't usually decrease just because it hasn't been used for a long time. I'm also not sure if any particular settings might be causing an increase.

**Reply by soul4kills** [reply]
600+mb. That's a lot. Mine is 199mb.

I had a memory leak issue as well. But I found out it was because I pointed flowlauncher to the python path that was installed on my system. I changed it and pointed it to the one included with flowlauncher, "pythonw.exe". And that fixed my memory leak issues.

That could be something to check.

**Reply by binronn** [reply]
Thanks, I'll try it

**Reply by soul4kills** [reply]
Have you tried disabling all plugins? To see if it's flow launcher itself?

Then try each plugin on by itself until you see the memory spike. Best way to find the culprit.

**Reply by jjw24** [reply]
> 600+mb. That's a lot. Mine is 199mb.
> 
> I had a memory leak issue as well. But I found out it was because I pointed flowlauncher to the python path that was installed on my system. I changed it and pointed it to the one included with flowlauncher, "pythonw.exe". And that fixed my memory leak issues.
> 
> That could be something to check.

What version of python and is it installed from Microsoft Store or via their official channel? Didn't know this would cause memory issues.

**Reply by soul4kills** [reply]
> What version of python and is it installed from Microsoft Store or via their official channel? Didn't know this would cause memory issues.

It was version 3.10. Any no it was not installed from the Microsoft Store. If I remember correctly, I think I also pointed it to "python.exe".  I never realized it wanted "pythonw.exe" because I never clicked the browse option. "pythonw.exe" is only ever mentioned when you click the browse button.

Pretty much had it installed like this for quite some time and was giving me lots of problems.

**Comment by binronn** [comment]
![image](https://github.com/user-attachments/assets/53f36552-898b-4480-ab7f-d2487498c0b3)
I deleted the plugins based on Python and Node.js, removed the paths for Python and Node.js, and restarted Flow Launcher. Right after the restart, the memory usage was already over 500 MB.

**Comment by binronn** [comment]
![image](https://github.com/user-attachments/assets/e36f0f26-f233-41e7-a8b4-5e60f2010355)

When I deleted the Cache folder, the History.jsonfile in the Settings folder, and the Logs, and then restarted Flow Launcher, the memory usage was only 50-70 MB during the first few seconds of startup. However, after a few seconds, the memory usage soared to over 380 MB again.

@jjw24

**Reply by taooceros** [reply]
I assume if you delete the cache folder and the memory drops, then the problem might be flow indexing too many programs in you computer.  Could you restart flow and see how many programs is loaded from cache?

**Reply by binronn** [reply]
> I assume if you delete the cache folder and the memory drops, then the problem might be flow indexing too many programs in you computer. Could you restart flow and see how many programs is loaded from cache?

![image](https://github.com/user-attachments/assets/d152d32a-27ee-41b4-8c15-bc5282cb320e)
![image](https://github.com/user-attachments/assets/5ac47c16-9939-4001-98e9-8001e652e481)
![image](https://github.com/user-attachments/assets/3c9b61b0-2448-46f6-b500-64fb9ed42c87)
![image](https://github.com/user-attachments/assets/e386fdec-e6c0-485f-9475-f0d7c82b9285)

Not much

**Reply by jjw24** [reply]
Disable the Programs plugin and see what the difference is.

**Reply by binronn** [reply]
> Disable the Programs plugin and see what the difference is.

I disabled all the plugins and restarted Flow Launcher, but there was no change.

**Comment by binronn** [comment]
![image](https://github.com/user-attachments/assets/24880191-9619-4049-a359-cde0af3a45f4)
I found an error in the log file during startup.

@jjw24

**Reply by taooceros** [reply]
Those are generally fine

**Comment by Garulf** [comment]
While it would be better to have this lower, it is also important to realize unused memory is wasted memory. 

What is really important is to make sure this isn't a memory leak of any sort. 

Does it stay stable at the amount you described or steadily increase over time?
